tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -1026,6 +1028,43 @@ public function test_remove_attribute_with_an_existing_attribute_name_removes_it
 		);
 	}
 
+	/**
+	 * @since 6.2.1
+	 * @ticket 58119
+	 * @covers WP_HTML_Tag_Processor::remove_attribute
+	 *
+	 * @dataProvider data_html_with_duplicated_attributes
+	 */
+	public function test_remove_attribute_with_duplicated_attributes_removes_all_of_them( $html_with_duplicate_attributes, $attribute_to_remove ) {
+		$p = new WP_HTML_Tag_Processor( $html_with_duplicate_attributes );
+		$p->next_tag();
+
+		$p->remove_attribute( $attribute_to_remove );
+		$this->assertSame( null, $p->get_attribute( $attribute_to_remove ), 'Failed to remove all copies of an attribute when duplicated in modified source.' );
+
+		// Recreate a tag processor with the updated HTML after removing the attribute.
+		$p = new WP_HTML_Tag_Processor( $p->get_updated_html() );
+		$p->next_tag();
+		$this->assertNull( $p->get_attribute( $attribute_to_remove ), 'Failed to remove all copies of duplicated attributes when getting updated HTML.' );
+	}
+
+	public function test_previous_duplicated_attributes_are_not_removed_on_successive_tag_removal() {
```

Comment:
Let's add a comment with `@since` and the other usual annotations to this test, too.

Reply:
thank you! will update